loki/transformations/tests/test_inline.py

```diff
@@ -706,6 +706,60 @@ def test_inline_member_routines_with_associate(frontend):
     assert len(assocs) == 2
 
 
+@pytest.mark.parametrize('frontend', available_frontends())
+def test_inline_member_routines_with_optionals(frontend):
+    """
+    Ensure that internal routines with optional arguments get
+    inlined as expected (esp. present instrinsics are correctly 
+    evaluated for all variables types)
+    """
+    fcode = """
+subroutine test_inline(ydxfu, klon)
+  implicit none
+
+  integer(kind=4), intent(in) :: klon, klev, kidia, kfdia, ktdia
+
+  type(txfu)              ,intent(inout)            :: ydxfu
+  type(mf_phys_out_type)  ,intent(in)               :: ydmf_phys_out
+ 
+  call member_rout (ydxfu%visicld, pvmin=ydmf_phys_out%visicld, psmax=1.0_8)
+
+  contains
+
+  subroutine member_rout (x, pvmin, pvmax, psmin, psmax)
+    
+    real(kind=8)         ,intent(inout)            :: x(1:klon)
+    real(kind=8)         ,intent(in)    ,optional  :: pvmin(1:klon)
+    real(kind=8)         ,intent(in)    ,optional  :: pvmax(1:klon)
+    real(kind=8)         ,intent(in)    ,optional  :: psmin
+    real(kind=8)         ,intent(in)    ,optional  :: psmax
+    type(toto), intent(in), optional :: dfmin
```

Comment:
dfmin should also be added to the interface of member_rout, and a corresponding import of `toto` should be added.